math/mathcore/inc/Fit/FitUtil.h

```diff
@@ -449,29 +449,16 @@ namespace FitUtil {
             return std::accumulate(objs.begin(), objs.end(), T{});
          };
 
-#ifndef R__USE_IMT
-         (void)nChunks;
-
-         // If IMT is disabled, force the execution policy to the serial case
-         if (executionPolicy == ::ROOT::EExecutionPolicy::kMultiThread) {
-            Warning("FitUtil::EvaluateChi2", "Multithread execution policy requires IMT, which is disabled. Changing "
-                                             "to ::ROOT::EExecutionPolicy::kSequential.");
-            executionPolicy = ::ROOT::EExecutionPolicy::kSequential;
-         }
-#endif
-
          T res{};
-         if (executionPolicy == ::ROOT::EExecutionPolicy::kSequential) {
-            ROOT::TSequentialExecutor pool;
-            res = pool.MapReduce(mapFunction, ROOT::TSeq<unsigned>(0, data.Size()/vecSize), redFunction);
-#ifdef R__USE_IMT
-         } else if (executionPolicy == ::ROOT::EExecutionPolicy::kMultiThread) {
-            ROOT::TThreadExecutor pool;
+         if (executionPolicy == ::ROOT::EExecutionPolicy::kMultiThread ||
+             executionPolicy == ::ROOT::EExecutionPolicy::kSequential) {
+            ROOT::TExecutor executor(executionPolicy);
             auto chunks = nChunks != 0 ? nChunks : setAutomaticChunking(data.Size() / vecSize);
-            res = pool.MapReduce(mapFunction, ROOT::TSeq<unsigned>(0, data.Size() / vecSize), redFunction, chunks);
-#endif
+            res = executor.MapReduce(mapFunction, ROOT::TSeq<unsigned>(0, data.Size() / vecSize), redFunction, chunks);
```

Comment:
In case of serial execution we did not use chunking before. Should not we continue doing this ? Or is it the same and there is no overhead in chunking?

Reply:
the chunking parameter is omitted in the sequential case.